provider/docker/docker_test.go

```diff
@@ -1091,3 +1091,72 @@ func TestDockerHasStickinessLabel(t *testing.T) {
 		})
 	}
 }
+
+func TestHasRequiredAPIVersion(t *testing.T) {
+	testCases := []struct {
+		desc            string
+		baseVersion     string
+		requiredVersion string
+		fallbackState   bool
+		expectedState   bool
+	}{
+		{
+			desc:            "base version > required version",
+			baseVersion:     "1.24",
+			requiredVersion: "1.12",
+			fallbackState:   false,
+			expectedState:   true,
+		},
+		{
+			desc:            "base version = required version",
+			baseVersion:     "1.12",
+			requiredVersion: "1.12",
+			fallbackState:   false,
+			expectedState:   true,
+		},
+		{
+			desc:            "base version < required version",
+			baseVersion:     "1.13",
+			requiredVersion: "1.26",
+			fallbackState:   true,
+			expectedState:   false,
+		},
+		{
+			desc:            "invalid base version and fallback state to true",
+			baseVersion:     "foo",
+			requiredVersion: "1.26",
+			fallbackState:   true,
+			expectedState:   true,
+		},
+		{
+			desc:            "invalid base version and fallback state to false",
+			baseVersion:     "foo",
+			requiredVersion: "1.26",
+			fallbackState:   false,
+			expectedState:   false,
+		},
+		{
+			desc:            "invalid required version and fallback state to true",
+			baseVersion:     "1.14",
+			requiredVersion: "bar",
+			fallbackState:   true,
+			expectedState:   true,
+		},
+		{
+			desc:            "invalid required version and fallback state to false",
+			baseVersion:     "1.14",
+			requiredVersion: "bar",
+			fallbackState:   false,
+			expectedState:   false,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+
+			state := hasRequiredAPIVersion2(test.baseVersion, test.requiredVersion, test.fallbackState)
```

Comment:
typo ? (`hasRequireAPIVersion2` ?)

Reply:
(and other question, it's not exported, should we really test that ? :stuck_out_tongue:)

Reply:
Ah, and there is https://github.com/moby/moby/blob/master/api/types/versions/compare.go too :angel: